common-lib/src/main/scala/com/gu/mediaservice/lib/auth/provider/Authentication.scala

```diff
@@ -0,0 +1,111 @@
+package com.gu.mediaservice.lib.auth.provider
+
+import com.gu.mediaservice.lib.argo.ArgoHelpers
+import com.gu.mediaservice.lib.argo.model.Link
+import com.gu.mediaservice.lib.auth.provider.Authentication.{ApiKeyAccessor, OnBehalfOfPrincipal, PandaUser, Principal}
+import com.gu.mediaservice.lib.auth.{ApiAccessor, Internal}
+import com.gu.mediaservice.lib.config.CommonConfig
+import com.gu.pandomainauth.model.{AuthenticatedUser, User}
+import com.gu.pandomainauth.service.Google2FAGroupChecker
+import play.api.libs.ws.WSRequest
+import play.api.mvc.Security.AuthenticatedRequest
+import play.api.mvc._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class Authentication(config: CommonConfig,
+                     providers: AuthenticationProviders,
+                     override val parser: BodyParser[AnyContent],
+                     override val executionContext: ExecutionContext)
+  extends ActionBuilder[Authentication.Request, AnyContent] with ArgoHelpers {
+
+  // make the execution context implicit so it will be picked up appropriately
+  implicit val ec: ExecutionContext = executionContext
+
+  val loginLinks = List(
+    Link("login", config.services.loginUriTemplate)
```

Comment:
Minor point but this should probably come from the provider rather than config now, assuming Kahuna does actually use it

Reply:
I've made this customisable.